frontend/ui/test/widgets/default_app_bar_test.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/material.dart';
+import 'package:tripmeout/widgets/default_app_bar.dart';
+
+void main() {
+  testWidgets('displays expected title', (WidgetTester tester) async {
+    await tester.pumpWidget(_InTestableContext(defaultAppBar));
+
+    expect(find.text('TripMeOut'), findsOneWidget);
+  });
+}
+
+typedef _AppBarProvider = Widget Function(BuildContext);
```

Comment:
Is this typedef for readability or does dart not have a way to declare an in-line function type?

Reply:
for readability.

Reply:
as far as I know it's like go's typedef

Reply:
+1 for readability. Sadly, looks like this is actually [more like](https://dart.dev/guides/language/language-tour#typedefs) Go type aliases. Go typedefs actually mint new mutually incompatible types (modulo explicit casts).